pkg/minishift/addon/manager/addon_manager.go

```diff
@@ -196,6 +196,33 @@ func (m *AddOnManager) ApplyAddOn(addOn addon.AddOn, context *command.ExecutionC
 	return nil
 }
 
+func (m *AddOnManager) RemoveAddOn(addOn addon.AddOn, context *command.ExecutionContext) error {
+	fmt.Print(fmt.Sprintf("-- Removing addon '%s':", addOn.MetaData().Name()))
+	context.AddToContext("addon-name", addOn.MetaData().Name())
+	defer context.RemoveFromContext("addon-name")
+
+	err := m.verifyRequiredVariablesInContext(context, addOn.MetaData())
+	if err != nil {
+		return err
+	}
+
+	oldDir, err := os.Getwd()
+	if err != nil {
+		return errors.Wrap(err, "Unable to remove addon due to failing IO operation")
+	}
+	defer os.Chdir(oldDir)
+
+	os.Chdir(addOn.InstallPath())
+	for _, c := range addOn.RemoveCommands() {
```

Comment:
+1 Much nicer now imo. I guess in the context of remove, there might be an even greater need to have commands which will not abort the overall execution in case an error occurs. But that's a different issue.